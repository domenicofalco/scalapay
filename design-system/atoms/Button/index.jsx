import PropTypes from "prop-types";
import styled from "styled-components";
import { base, black } from "fonts";
import { grey900, lightBlue, beige } from "colors";

const common = {
  fontSize: base,
  color: grey900,
  fontWeight: black,
  textTransform: "uppercase",
  border: `2px solid ${grey900}`,
  borderRadius: "30px",
  padding: "16px",
  cursor: "pointer"
};

const primary = styled.button`
  background: ${lightBlue};
  ${common};
`;

const secondary = styled.button`
  background: ${beige};
  ${common}
`;

const roundIcon = styled.button`
  background: ${beige};
  border: 2px solid ${grey900};
  color: ${grey900};
  font-size: ${base};
  cursor: pointer;
  width: 30px;
  height: 30px;
  border-radius: 100%;
`;

const Button = ({ children, variant, ...rest }) => {
  const El = Button.variants[variant] || primary;
  return <El {...rest}>{children}</El>;
};

Button.variants = {
  primary,
  secondary,
  roundIcon
};

Button.defaultProps = {
  variant: "primary"
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.string.isRequired
};

export default Button;
