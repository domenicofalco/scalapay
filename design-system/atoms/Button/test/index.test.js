import React from "react";
import { cleanup, render } from "@testing-library/react";
import Button from "../";

afterEach(cleanup);

describe("atoms => Button", () => {
  it("It should render 'primary' variant as default prop is missing or wrong", () => {
    const { getByText } = render(<Button type="button">Button</Button>);
    const component = getByText(/Button/i);
    expect(component.dataset.variant).toBe("primary");
  });

  it("It should render 'primary' variant", () => {
    const { getByText, asFragment } = render(
      <Button variant="primary" type="button">
        Button
      </Button>
    );
    const component = getByText(/Button/i);
    expect(component.dataset.variant).toBe("primary");
    expect(asFragment()).toMatchSnapshot();
  });

  it("It should render 'secondary' variant", () => {
    const { getByText } = render(
      <Button variant="secondary" type="button">
        Button
      </Button>
    );
    const component = getByText(/Button/i);
    expect(component.dataset.variant).toBe("secondary");
  });

  it("It should render 'roundIcon' variant", () => {
    const { getByText } = render(
      <Button variant="roundIcon" type="button">
        Button
      </Button>
    );
    const component = getByText(/Button/i);
    expect(component.dataset.variant).toBe("roundIcon");
  });
});
