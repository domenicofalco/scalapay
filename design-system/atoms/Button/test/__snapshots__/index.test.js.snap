// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms => Button It should render 'primary' variant 1`] = `
<DocumentFragment>
  <button
    class="Button__primary-sc-6egjvv-0 kmwFgP"
    data-variant="primary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
